lib/wallet/src/Cardano/Wallet/Write/Tx/Gen.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE ExplicitNamespaces #-}
+{-# LANGUAGE PatternSynonyms #-}
+module Cardano.Wallet.Write.Tx.Gen
+    ( genDatum
+    , genData
+    , genDatumHash
+    )
+    where
+
+import Prelude
+
+import Cardano.Ledger.Alonzo.Data
+    ( Data (..), dataToBinaryData)
+import Cardano.Wallet.Write.Tx
+    ( type Datum, type DatumHash, pattern Datum, pattern DatumHash, pattern NoDatum, datumHashFromBytes, BinaryData )
```

Comment:
Not thrilled about the `pattern Datum, type Datum` + `LANGUAGE ExplicitNamespaces, PatternSynonyms` requirement... But the only other alternatives would be:
1. Export a `Datum` type without pre-applied `BabbageEra StandardCrypto`
2. Create our own `data Datum = Datum .. | DatumHash ... | NoDatum` and convert to/from `Alonzo.Datum`

🤔

Reply:
The `type Datum` seems to break `stylish-haskell`, so I can't leave it as is at least.

Reply:
I went with defining the `Write.Tx` functions in terms of e.g. `Datum LatestLedgerEra`. In the future we may want them to operate on `IsRecentEra era. era`, but let's deal with that... in the future.